osu.Game/Screens/Edit/Timing/TapTimingControl.cs

```diff
@@ -192,13 +210,24 @@ private void reset()
             editorClock.Seek(selectedGroup.Value.Time);
         }
 
+        private List<HitObject> hitObjectsInTimingRange(EditorBeatmap beatmap, ControlPointGroup selectedGroup)
+        {
+            // If the first group, we grab all hitobjects prior to the next, if the last group, we grab all remaining hitobjects
+            double firstGroupTime = beatmap.ControlPointInfo.Groups.Any(x => x.ControlPoints.Any(y => y is TimingControlPoint) && x.Time < selectedGroup.Time) ? selectedGroup.Time : double.MinValue;
+            double nextGroupTime = beatmap.ControlPointInfo.Groups.FirstOrDefault(x => x.ControlPoints.Any(y => y is TimingControlPoint) && x.Time > selectedGroup.Time)?.Time ?? double.MaxValue;
+
+            return beatmap.HitObjects.Where(x => Precision.AlmostBigger(x.StartTime, firstGroupTime) && Precision.DefinitelyBigger(nextGroupTime, x.StartTime)).ToList();
```

Comment:
- what's with the `Precision.AlmostBigger()` calls here?
- why is the second check using start time and not end time?

Reply:
Precision calls were due to when using >= and >, that notes placed at the start/end would behave unexpectedly.

The second check still uses start time as I felt this behaviour was more expected, as using end time would break snapping behaviour.
I'll check stable next time I get the chance to make sure it matches, however.

Reply:
> as using end time would break snapping behaviour

What does this mean?

Reply:
What I presume you meant was changing `Precision.DefinitelyBigger(nextGroupTime, x.StartTime)` to `Precision.DefinitelyBigger(nextGroupTime, x.EndTime)`

Which would break snapping behaviour, as the start time would no longer be on a snapping point if a hit object with duration had a start time before the next timing point, and a end time after the next timing point.

Reply:
Well if that is a concern then I argue the *first* range check is wrong and should be checking end time.

Reply:
I am a bit confused, hmm...
I will assume that the two numbers I have for startTime and endTime are without issue.

For deciding which hitobjects to adjust, we want to take all hit objects who are snapped to some point in the above range, inclusive on the startTime, and exclusive on the endTime. 

To do this, we check each hit object's start time, and check if it's in the above range. This should be an equivalent check.

To do this check, we don't care about a hit object's end time as it plays no part in the snapping point.


As an example, if we adjust the settings for the 90bpm node, given the above we should expect it to change, even with it's end node in the next range, so it maintains snapped to the same beat divisor.
If we adjust settings for the 100bpm node, however, despite the end time being in range, the slider should not be adjusted as this would make the slider no longer snapped to the same beat divisor.
<img width="292" alt="image" src="https://github.com/ppy/osu/assets/7444073/51bdebf0-3a94-4f07-992f-f2b1d7fce39c">

I feel this is the expected behaviour? Unless I misunderstand what you're saying, implementing either of your changes would break this behaviour.

Reply:
First of all the slider above already looks unsnapped to me.

My expectation would be that regardless of whether of which of the two timing points on the screenshot above is modified, *both* ends of the object should remain snapped to the *same* beat they were previously snapped to. I imagine with the implementation that you currently have, if you edit the BPM on the second control point, the object's end will unsnap (because the object will be excluded from being adjusted, while it *really* should have its duration shortened or lengthened, depending on which way BPM of the second point changed).

Is this making more sense?

Reply:
Oh, hmm... I think I get your point now, although I do still disagree with it.

Current behaviour on master doesn't have any snapping for sliders crossing over a timing point in the new range.
<img width="636" alt="image" src="https://github.com/ppy/osu/assets/7444073/8a4dd5b8-f44a-4777-a5d2-ffe0e42f3235">

Given this, and my opinion that the starting time is much more important to remain snapped, I feel the most expected outcome is still as implemented.
To implement the end node snapping as you're suggesting... It might be possible to do by adjusting slider velocity, I suppose, but that feels jank, and I have the feeling will still have some snapping behaviour break with repeats.

Unless you're saying slider velocity should change to match the timing section the section the slider is in? in which case, this would be out of scope for this PR, I think.

But perhaps feedback from people who actually use the editor more than me might be a fair idea in this regard with what to do as well, don't know many though... But I can still ask those I know, I guess.

Reply:
Checked against stable. Current implementation should match stable with how slider behaviour crossing a timing point works.

Also a video of how my implementation works, which might help.

https://github.com/ppy/osu/assets/7444073/69a2291e-4a25-47ed-a5ab-6fc9413022e9

As for feedback so far, is just looking for parity namely. But also see frequently asked to have other control points adjust to changes, such that they maintain being snapped.

Have also had someone request that these settings are saved per control point too.

Not sure what to think about the latter suggestion, but given the former is fairly frequently requested so far, and I can see the use, I think it would be worthwhile to implement (as a secondary toggle), but let me know your thoughts.

Reply:
> I think it would be worthwhile to implement (as a secondary toggle)

No this should not be a toggle. If mappers are used to the current behaviour then I can concede on this point I guess.

Reply:
Hmm, okay. I'll keep it as is then for now. 

Once I get some more feedback, I'll see about creating a discussion on the other requests if I think is warranted, I suppose, as well.